lib/storage/providers/LocalForage.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * @file
+ * The storage provider based on localforage allows us to store most anything in its
+ * natural form in the underlying DB without having to stringify or de-stringify it
+ */
+
+import localforage from 'localforage';
+import _ from 'underscore';
+import lodashMerge from 'lodash/merge';
+
+localforage.config({
+    name: 'OnyxDb'
+});
+
+const provider = {
+    multiGet(keys) {
+        const pairs = [];
+        const remainingKeys = new Set(keys);
+
+        return localforage.iterate((value, key) => {
+            if (remainingKeys.has(key)) {
+                remainingKeys.delete(key);
+                pairs.push([key, value]);
+            }
+
+            // Exist early (don't iterate all keys) if we've mapped everything
+            if (remainingKeys.size === 0) {
+                return true;
+            }
+
+            return undefined;
+        })
+            .then(() => pairs);
+    },
+    multiMerge(pairs) {
+        const tasks = pairs.map(([key, delta]) => this.getItem(key)
+            .then((existing) => {
+                let merged = delta;
+
+                if (existing) {
+                    if (_.isObject(existing)) {
```

Comment:
I don't see it documented on AsyncStorage but I think this is the same behavior as AsyncStorage's `multiMerge` 

The strategy as I remember is replaces or add keys, with objects it's a no brainer new keys are added existing keys get replaced, but for arrays it's always a bummer
I think this is what happens (and `lodashMerge` works that way) - if we imagine an array with 3 keys as object it will be an object like `{ '0': 'val', '1': 'val', '2': 'val' }` so you can see where merging an array with 2 or 4 keys would get you